src/Agent.Worker/ContainerOperationProvider.cs

```diff
@@ -567,6 +591,34 @@ private async Task<List<string>> ExecuteCommandAsync(IExecutionContext context,
         }
 #endif
 
+#if OS_LINUX
+        private async Task<bool> IsMobyInstalled(IExecutionContext context)
+        {            
+            try
+            {
+                using (var processInvoker = HostContext.CreateService<IProcessInvoker>())
+                {
+                    await processInvoker.ExecuteAsync(
+                                            workingDirectory: HostContext.GetDirectory(WellKnownDirectory.Work),
+                                            fileName: "dpkg",
+                                            arguments: "-S moby",
+                                            environment: null,
+                                            requireExitCodeZero: true,
+                                            outputEncoding: null,
+                                            cancellationToken: CancellationToken.None);
+                    context.Debug($"Moby package installed.");
+                    return true;
```

Comment:
Does `dpkg` return with a non zero exit code when the search results are empty?
If it doesn't, then this function would always return true.